lib/livebook/zta/tailscale.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Livebook.ZTA.Tailscale do
+  @doc """
+  To integrate Tailscale authentication with Livebook,
+  set the `LIVEBOOK_IDENTITY_PROVIDER` environment variable to `tailscale:tailscale-socket-path`.
+
+  If you want to access the Livebook on the same machine as you are hosting it on,
+  you will also need to set the LIVEBOOK_IP variable to your Tailscale IP.
+
+  To do both of these things, run
+
+  ```bash
+  LIVEBOOK_IP=$(tailscale ip | head -1 | tr -d '\n') LIVEBOOK_IDENTITY_PROVIDER=tailscale:/var/run/tailscale/tailscaled.sock livebook server
+  ```
+
+  See https://tailscale.com/blog/tailscale-auth-nginx/ for more information
+  on how Tailscale authorization works.
+
+  ## MacOS
+
+  On MacOS, when Tailscale is installed via the Mac App Store, no unix socket is exposed.
+  Instead, a TCP port is made available, protected via a password, which needs to be located.
+  Tailscale itself uses lsof for this. This method is replicated in the bash script below,
+  which will start Livebook with your Tailscale IP and correct port and password.
+
+  ```bash
+  #!/bin/bash
+  addr_info=$(lsof -n -a -c IPNExtension -F | sed -n 's/.*sameuserproof-\([[:digit:]]*-.*\).*/\1/p')
+  port=$(echo "$addr_info" | cut -d '-' -f 1)
+  pass=$(echo "$addr_info" | cut -d '-' -f 2)
+  LIVEBOOK_IP=$(exec $(ps -xo comm | grep MacOS/Tailscale$) ip | head -1 | tr -d '\n') LIVEBOOK_IDENTITY_PROVIDER=tailscale:http://:$pass@127.0.0.1:$port livebook server
```

Comment:
> `LIVEBOOK_IP=$(exec ...)`

in my local testing (against Mac App Store-installed tailscale) it was enough to set `LIVEBOOK_IP` to `$(tailscale ip --1)` so maybe we show that here too?

As an aside, @josevalim: would it be possible to just set `LIVEBOOK_IDENTITY_PROVIDER=tailscale` and do the rest in Elixir e.g. when the app boots? I think it would be a much better UI if users don't have to copy paste a bunch of shell commands.

Similarly, perhaps we don't need to require users to set `LIVEBOOK_IP` but can do that ourselves internally.

That being said I think it is totally fine to start with something really simple if verbose and improve this as more people use the feature.

Reply:
Let's update the examples to use `tailscale ip --1` but require those env vars to be set. I don't want to rely on `lsof` for macos and the setting of `LIVEBOOK_IP` happens at a very different place which I am not sure it is worth coupling.

Reply:
>in my local testing (against Mac App Store-installed tailscale) it was enough to set LIVEBOOK_IP to $(tailscale ip --1) so maybe we show that here too?

It works if you have a Tailscale alias in your shell, I got this command [here](https://github.com/tailscale/tailscale/issues/2553#issuecomment-912824816) because it (should) work on all types of MacOS installs without requiring an alias to be set.

Reply:
Ah, good point, the CLI doesn’t get installed by default. Fine by me.

Reply:
Or in our instructions we do the alias. I think that’s better. And if we can somehow get the password out of the cli with less code that’s even better. But no strong opinion.